python_modules/dagster/dagster/_core/definitions/data_time.py

```diff
@@ -502,6 +502,25 @@ def get_current_data_time(
 
         return min(cast(AbstractSet[datetime.datetime], data_times), default=None)
 
+    def _get_source_data_time(
+        self, asset_key: AssetKey, current_time: datetime.datetime
+    ) -> Optional[datetime.datetime]:
+        latest_record = self.instance_queryer.get_latest_materialization_or_observation_record(
+            AssetKeyPartitionKey(asset_key)
+        )
+        if latest_record is None:
+            return None
+        if latest_record.asset_observation is None:
+            raise DagsterInvariantViolationError(
+                "Cannot calculate data time for source asset with materialization"
+            )
+        # handle case where data_time isn't set
```

Comment:
Is this comment accurate? Is this not the case where it _is_ set?

Reply:
Meant as a TODO